editor/src/core/commenting/comment-hooks.tsx

```diff
@@ -261,3 +261,79 @@ export function useUnresolvedThreads() {
     threads: threads.threads.filter((t) => t.metadata.resolved !== true),
   }
 }
+
+export function useSetThreadReadStatusOnMount(thread: ThreadData<ThreadMetadata> | null) {
+  const setThreadReadStatus = useSetThreadReadStatus()
+
+  React.useEffect(() => {
+    if (thread == null) {
+      return
+    }
+
+    setThreadReadStatus(thread.id, 'read')
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []) // only run it once on mount, because opening the popup means reading the thread, no dependencies added
+}
+
+export function useMyThreadReadStatus(thread: ThreadData<ThreadMetadata> | null): ThreadReadStatus {
+  const self = useSelf()
+  return useStorage((store) => {
+    if (thread == null) {
+      return 'unread'
+    }
+    const statusesForThread = store.userReadStatusesByThread[thread.id]
+    if (statusesForThread == null) {
+      return 'unread'
+    }
+    return statusesForThread[self.id] === true ? 'read' : 'unread'
```

Comment:
If we're never setting this to `false` then we should probably just always return `'read'` at this point, since really what we're doing here is using the object to store a flag per user that has read the thread. If we start using `false` to represent that the user hasn't read the thread then that's added complexity for no gain